backend/controllers/userDetails.test.js

```diff
@@ -65,116 +66,127 @@ describe('user detail factory', () => {
-      ...expectedUserDetails,
-      groups: [{ groupName: 'groupName2', groupCode: 'groupCode2', showRemove: true }],
-      showEnableDisable: false,
-      showExtraUserDetails: false,
-      showGroups: false,
+    it('should copy the search results url through from the session', async () => {
+      const searchResultsReq = {
+        ...req,
+        session: { searchResultsUrl: '/some-url' },
+      }
+      getUserRolesAndGroupsApi.mockResolvedValue([userStub, rolesStub, groupsStub])
+      await userDetails.index(searchResultsReq, { render })
+      expect(render).toBeCalledWith('userDetails.njk', {
+        ...expectedUserDetails,
+        searchResultsUrl: '/some-url',
+      })
     })
-  })
 
-  it('should copy the search results url through from the session', async () => {
-    const searchResultsReq = {
-      ...req,
-      session: { searchResultsUrl: '/some-url' },
-    }
-    getUserRolesAndGroupsApi.mockResolvedValue([userStub, rolesStub, groupsStub])
-    await userDetails.index(searchResultsReq, { render })
-    expect(render).toBeCalledWith('userDetails.njk', {
-      ...expectedUserDetails,
-      searchResultsUrl: '/some-url',
+    it('should call getUserRolesAndGroupsApi with maintain admin flag set to false', async () => {
+      getUserRolesAndGroupsApi.mockResolvedValue([userStub, rolesStub, groupsStub])
+      const locals = { user: { maintainAuthUsers: true } }
+      await userDetails.index(req, { render: jest.fn(), locals })
+      expect(getUserRolesAndGroupsApi).toBeCalledWith(locals, '00000000-aaaa-0000-aaaa-0a0a0a0a0a0a', false, true)
     })
-  })
 
-  it('should call getUserRolesAndGroupsApi with maintain admin flag set to false', async () => {
-    getUserRolesAndGroupsApi.mockResolvedValue([userStub, rolesStub, groupsStub])
-    const locals = { user: { maintainAuthUsers: true } }
-    await userDetails.index(req, { render: jest.fn(), locals })
-    expect(getUserRolesAndGroupsApi).toBeCalledWith(locals, '00000000-aaaa-0000-aaaa-0a0a0a0a0a0a', false, true)
-  })
+    it('should call getUserRolesAndGroupsApi with maintain admin flag set to true', async () => {
+      getUserRolesAndGroupsApi.mockResolvedValue([userStub, rolesStub, groupsStub])
+      const locals = { user: { maintainAccessAdmin: true } }
+      await userDetails.index(req, { render: jest.fn(), locals })
+      expect(getUserRolesAndGroupsApi).toBeCalledWith(locals, '00000000-aaaa-0000-aaaa-0a0a0a0a0a0a', true, false)
+    })
 
-  it('should call getUserRolesAndGroupsApi with maintain admin flag set to true', async () => {
-    getUserRolesAndGroupsApi.mockResolvedValue([userStub, rolesStub, groupsStub])
-    const locals = { user: { maintainAccessAdmin: true } }
-    await userDetails.index(req, { render: jest.fn(), locals })
-    expect(getUserRolesAndGroupsApi).toBeCalledWith(locals, '00000000-aaaa-0000-aaaa-0a0a0a0a0a0a', true, false)
+    it('uses default search results url when nothing provided through session', async () => {
```

Comment:
... apart from this one that I've added to show what happens when no urls saved in the session